File: files/200-299/244/CH-244.py
import pandas as pd
import re

path = "200-299/244/CH-244 Column Splitting.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="D:E", skiprows=1, nrows=6)

def split_by_sec_del(text):
    if pd.isna(text): return [text, None]
    h = [m.start() for m in re.finditer("-", str(text))]
    s = [m.start() for m in re.finditer("/", str(text))]
    if sum(len(x) >= 2 for x in [h, s]) == 0: return [text, None]
    if h and s and len(h) >= 2 and len(s) >= 2:
        i = min(h[1], s[1])
    elif len(h) >= 2:
        i = h[1]
    else:
        i = s[1]
    return [text[:i], text[i+1:]]

split_cols = input.iloc[:, 0].apply(split_by_sec_del)
result = pd.DataFrame(split_cols.tolist(), columns=["ID.1", "ID.2"])

print(result.equals(test)) # True

File: files/200-299/244/CH-244.R
library(tidyverse)
library(readxl)

path = "files/200-299/244/CH-244 Column Splitting.xlsx"
input = read_excel(path, range = "B2:B7")
test = read_excel(path, range = "D2:E7")

split_by_sec_del = function(text) {
  hyph_loc = str_locate_all(text, "-")[[1]]
  sl_loc = str_locate_all(text, "\\/")[[1]]

  if (nrow(hyph_loc) < 2 & nrow(sl_loc) < 2) {
    return(text)
  } else if (nrow(hyph_loc) >= 2 & nrow(sl_loc) >= 2) {
    split_loc <- min(hyph_loc[2, 1], sl_loc[2, 1])
  } else if (nrow(hyph_loc) >= 2) {
    split_loc <- hyph_loc[2, 1]
  } else {
    split_loc <- sl_loc[2, 1]
  }
  first_part = str_sub(text, 1, split_loc - 1)
  second_part = str_sub(text, split_loc + 1)
  return(c(first_part, second_part))
}

result = input %>%
  mutate(ID = map(ID, split_by_sec_del)) %>%
  unnest_wider(ID, names_sep = ".")

all.equal(result, test, check.attributes = FALSE)
# TRUE
